site/src/comparison.rs

```diff
@@ -421,61 +422,103 @@ impl ComparisonSummary {
             .unwrap();
         }
     }
+}
 
-    /// Writes a Markdown table containing summary of relevant results.
-    pub fn write_summary_table(&self, result: &mut String) {
-        use std::fmt::Write;
+pub type BenchmarkMap = HashMap<Benchmark, BenchmarkData>;
 
-        fn render_stat<F: FnOnce() -> Option<f64>>(count: usize, calculate: F) -> String {
-            let value = if count > 0 {
-                calculate()
-            } else {
-                None
-            };
-            value.map(|value| format!("{value:.1}%")).unwrap_or_else(|| "N/A".to_string())
-        }
+/// Writes a Markdown table containing summary of relevant results.
+pub fn write_summary_table(
+    primary: &ComparisonSummary,
+    secondary: &ComparisonSummary,
+    result: &mut String,
+) {
+    use std::fmt::Write;
 
-        writeln!(
-            result,
-            r#"| | Regressions 😿 | Improvements 🎉 | All relevant changes |
-|:---:|:---:|:---:|:---:|"#
-        )
-        .unwrap();
-        writeln!(
-            result,
-            "| count[^1] | {} | {} | {} |",
-            self.num_regressions,
-            self.num_improvements,
-            self.num_regressions + self.num_improvements
-        )
-        .unwrap();
-        writeln!(
-            result,
-            "| mean[^2] | {} | {} | {:.1}% |",
-            render_stat(self.num_regressions, || Some(self.arithmetic_mean_of_regressions())),
-            render_stat(self.num_improvements, || Some(self.arithmetic_mean_of_improvements())),
-            self.arithmetic_mean_of_changes()
-        )
-            .unwrap();
+    fn render_stat<F: FnOnce() -> Option<f64>>(count: usize, calculate: F) -> String {
+        let value = if count > 0 { calculate() } else { None };
+        value
+            .map(|value| format!("{value:.1}%"))
+            .unwrap_or_else(|| "N/A".to_string())
+    }
+
+    writeln!(
+        result,
+        r#"| | Regressions 😿 <br> (primary) | Regressions 😿 <br> (secondary) | Improvements 🎉 <br> (primary) | Improvements 🎉 <br> (secondary) | All relevant changes |
```

Comment:
I think if we keep the "All relevant changes" column it should just be the primary benchmarks.

Reply:
Okay, I think I agree. But before I change it, @Mark-Simulacrum @nnethercote @lqd what are your thoughts on this?

Reply:
Having it summarize the primary benchmarks also seems sensible to me, to de-emphasize the secondary benchmark results (and their data is still there next to it anyways), but I myself mostly focus on the magnitude of regressions and improvements individually rather than the complete aggregate results.